core/src/main/scala/org/apache/spark/memory/MemoryManager.scala

```diff
@@ -200,6 +201,12 @@ private[spark] abstract class MemoryManager(
    */
   final def offHeapStorageMemoryUsed: Long = offHeapStorageMemoryPool.memoryUsed
 
+  /**
+   *  If the system isn't procfsBased the process tree metrics' values will be -1,
+   *  meaning not available
+   */
+  final val pTreeInfo: ProcessTreeMetrics = new ProcfsBasedSystems
```

Comment:
the `MemoryManager` is not a very good spot for this.  You could put it in ExecutorMetricType, or just have an `object ProcfsBasedSystems`

Reply:
will find a better place for it. I jus put it there since this also gathering memory info